pkg/processor/trigger/kafka/test/kafka_test.go

```diff
@@ -158,6 +183,115 @@ func (suite *testSuite) TestReceiveRecords() {
 		suite.publishMessageToTopic)
 }
 
+func (suite *testSuite) TestExplicitAck() {
+
+	topic := "myNewTopic"
+	shardID := int32(1)
+	functionName := "explicitacker"
+	functionPath := path.Join(suite.GetTestFunctionsDir(),
+		"python",
+		"kafka-explicit-ack",
+		"explicitacker.py")
+
+	// create new topic
+	createTopicsResponse, err := suite.broker.CreateTopics(&sarama.CreateTopicsRequest{
+		TopicDetails: map[string]*sarama.TopicDetail{
+			topic: {
+				NumPartitions:     suite.NumPartitions,
+				ReplicationFactor: 1,
+			},
+		},
+	})
+	suite.Require().NoError(err, "Failed to create topic")
+
+	suite.Logger.InfoWith("Created topic",
+		"topic", topic,
+		"createTopicResponse", createTopicsResponse)
+
+	// create explicit ack function
+
+	createFunctionOptions := suite.GetDeployOptions(functionName, functionPath)
+	createFunctionOptions.FunctionConfig.Spec.Build.Commands = []string{"pip install nuclio-sdk"}
+	createFunctionOptions.FunctionConfig.Spec.Platform = functionconfig.Platform{
+		Attributes: map[string]interface{}{
+			"network": suite.BrokerContainerNetworkName,
+		},
+	}
+
+	// configure kafka trigger with explicit ack enabled
+	createFunctionOptions.FunctionConfig.Spec.Triggers = map[string]functionconfig.Trigger{
+		"my-kafka": {
+			Kind: "kafka-cluster",
+			URL:  fmt.Sprintf("%s:9090", suite.brokerContainerName),
+			Attributes: map[string]interface{}{
+				"topics":        []string{topic},
+				"consumerGroup": functionName,
+				"initialOffset": suite.initialOffset,
+			},
+			WorkerTerminationTimeout: "5s",
+			ExplicitAckMode:          functionconfig.ExplicitAckModeEnable,
+		},
+		"my-http": {
+			Kind: "http",
+			//URL:  httpURL,
+			Attributes: map[string]interface{}{},
+		},
+	}
+
+	// set worker allocation mode to static
+	createFunctionOptions.FunctionConfig.Meta.Annotations = map[string]string{
+		"nuclio.io/kafka-worker-allocation-mode": string(partitionworker.AllocationModeStatic),
+	}
+
+	// deploy function
+	suite.DeployFunction(createFunctionOptions, func(deployResult *platform.CreateFunctionResult) bool {
+		suite.Require().NotNil(deployResult, "Unexpected empty deploy results")
+
+		var err error
+
+		// publish 10 messages to the topic
+		for i := 0; i < 10; i++ {
+			err = suite.publishMessageToTopicOnSpecificShard(topic, fmt.Sprintf("message-%d", i), shardID)
+			suite.Require().NoError(err, "Failed to publish message")
+		}
+
+		// ensure queue size is 10
+		suite.Logger.Debug("Getting current queue size")
+		queueSize := suite.getQueueSize(deployResult.Port)
+		suite.Require().Equal(queueSize, 10, "Queue size is not 10")
+
```

Comment:
hmm, there might be a gap between when you publish a message to the fact it was queued on your function. I suggest wrapping the get queue size function with something like `waitForFunctionQueueSize(expectedQueueSize, timeout)`